src/main/java/mate/academy/service/impl/CinemaHallServiceImpl.java

```diff
@@ -19,11 +19,24 @@ public CinemaHall add(CinemaHall cinemaHall) {
 
     @Override
     public CinemaHall get(Long id) {
-        return cinemaHallDao.get(id).get();
+        return cinemaHallDao.get(id).orElseThrow(
+                () -> new RuntimeException("cinemaHall not found"));
     }
 
     @Override
     public List<CinemaHall> getAll() {
         return cinemaHallDao.getAll();
     }
+
+    @Override
+    public Boolean update(CinemaHall cinemaHall) {
+        CinemaHall currentCinemaHall = get(cinemaHall.getId());
+        cinemaHallDao.update(cinemaHall);
+        return currentCinemaHall.equals(get(cinemaHall.getId()));
```

Comment:
The `update` method returns a boolean by comparing the updated entity with the one retrieved from the database. This approach might not be reliable if the entity has fields that are not persisted or if the database automatically modifies some fields (e.g., timestamps). Consider returning a boolean based on the success of the update operation itself, rather than comparing objects.